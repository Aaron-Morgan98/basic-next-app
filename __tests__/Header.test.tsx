import { render } from "@testing-library/react";
import Header from "../src/app/components/header";

// Mock the `useTranslations` hook
jest.mock("next-intl", () => ({
  useTranslations: () => (key: string) => key === "TITLE" ? "Mocked Title" : key,
}));

describe("Header component", () => {
  it("renders correctly and matches snapshot", () => {
    const { asFragment } = render(<Header />);
    expect(asFragment()).toMatchSnapshot();
  });

  it("displays the correct title text", () => {
    const { getByText } = render(<Header />);
    // Assert that the title contains the mocked text
    expect(getByText("Mocked Title")).toBeInTheDocument();
  });
});
