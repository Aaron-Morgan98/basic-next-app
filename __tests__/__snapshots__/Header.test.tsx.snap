// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header component renders correctly and matches snapshot 1`] = `
<DocumentFragment>
  <div
    class="headerContainer"
  >
    <div
      class="title"
    >
      <h1>
         Mocked Title 
      </h1>
    </div>
  </div>
</DocumentFragment>
`;
